src/basic/path-util.c

```diff
@@ -683,64 +737,36 @@ int find_executable_full(const char *name, const char *root, bool use_path_envva
                 return 0;
         }
 
+        if (exec_search_paths) {
+                char **element;
+                STRV_FOREACH(element, exec_search_paths) {
+                        last_error = find_executable_from_path(name, root, *element, ret_filename, ret_fd);
+                        if (last_error == 0)
+                                return 0;
+                }
+                return last_error;
```

Comment:
hmm? last_error is unnecessary, you might as well use r here if you do it like this?